src/importlinter/domain/contract.py

```diff
@@ -22,17 +23,23 @@ def _populate_fields(self) -> None:
         Raises:
             InvalidContractOptions if the contract options could not be matched to the fields.
         """
+        contract_options = copy.deepcopy(self.contract_options)
+
         errors = {}
         for field_name in self.__class__._get_field_names():
             field = self.__class__._get_field(field_name)
 
             try:
-                raw_data = self.contract_options[field_name]
+                raw_data = contract_options.pop(field_name)
             except KeyError:
                 if field.required:
                     errors[field_name] = "This is a required field."
                 else:
-                    setattr(self, field_name, None)
+                    # field.default is None if it isn't declared
+                    if callable(field.default):
```

Comment:
NIP: I don't think it's required, in my opinion, it's clearer to use immutable objects as `default` arguments